version: '2.7'

services:
  web:
    build:
      context: .
      dockerfile: packaging/docker/AuthService/Dockerfile
    command: >
      sh -c "echo 'Running in development environment!' &&
             python -m tornado.autoreload main.py"
    volumes:
      - ./src/AuthService:/AuthService
    env_file:
      - .env
    ports:
      - 8888:8888
    depends_on:
      - postgres_db
    restart: on-failure

  postgres_db:
    image: postgres:13.0
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    env_file:
      - .env
    ports:
    - 5432:5432
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -d ${DB_NAME} -U ${DB_USER}"]
      timeout: 10s
      interval: 1s
      retries: 20

volumes:
  postgres_data:
